lib/geminabox/gem_list_merge.rb

```diff
@@ -1,23 +1,38 @@
-module Geminabox
-  class GemListMerge
-    attr_accessor :list
-
-    IGNORE_DEPENDENCIES = 0..-2
+require "set"
 
-    def self.from(*lists)
-      lists.map{|list| new(list)}.inject(:merge)
+module Geminabox
+  module GemListMerge
+    def self.merge(local_gem_list, remote_gem_list, strategy:)
+      strategy_for(strategy).merge(local_gem_list, remote_gem_list)
     end
 
-    def initialize(list)
-      @list = list
+    def self.strategy_for(strategy)
+      case strategy
+      when :local_gems_take_precedence_over_remote_gems
+        LocalGemsTakePrecedenceOverRemoteGems
+      when :combine_local_and_remote_gem_versions
+        CombineLocalAndRemoteGemVersions
+      else
+        raise ArgumentError, "Merge strategy must be :local_gems_take_precedence_over_remote_gems (default) or :merge_local_and_remote_gem_versions"
```

Comment:
If the code is going to force this setting to be set in the configuration then that should be stated in the README.

Reply:
This has a default value. If you choose to set a disallowed value, it will fail.